### PR TITLE
Fix JSON_QUERY function with multiple numeric paths

### PR DESCRIPTION
**Pull request by KevinyhZou**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix the JSON_QUERY function can not parse the json string while path have multiple sub-paths split by `.` or `[]`, and when one sub-path is numberic, then the query will be failed.

like in the query 
SELECT JSON_QUERY('{"123":{"123":"abcd"}}', '$.123.123'), 
SELECT JSON_QUERY('{"abc":{"123":"abcd"}, '$.abc.123'}'
SELECT JSON_QUERY('{"123":[0]}', '$.123[0]')

when execute them, we would encounter the exceptions

```
DB::Exception: Unable to parse JSONPath: While processing JSON_QUERY(... ...). (BAD_ARGUMENTS)
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse-ci-1** [comment]
*This is an automated comment for commit 97d73c1f767ff06ff4f9ee4c537fc99cdc0e9e1c with description of existing statuses. It's updated for the latest CI running*

[❌ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57096/97d73c1f767ff06ff4f9ee4c537fc99cdc0e9e1c/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6956565576">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57096/97d73c1f767ff06ff4f9ee4c537fc99cdc0e9e1c/ci_running.html">❌ failure</a></td></tr>
<tbody>
</table>

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp [hunk 1]
```diff
@@ -58,14 +57,16 @@ bool ParserJSONPathMemberAccess::parseImpl(Pos & pos, ASTPtr & node, Expected &
             member_name = std::make_shared<ASTIdentifier>(String(last_begin, pos->end));
             ++pos;
         }
-        else if (!pos.isValid() && pos->type == TokenType::EndOfStream)
+        else if (pos.isValid() && (pos->type == TokenType::Dot || pos->type == TokenType::OpeningSquareBracket))
         {
             member_name = std::make_shared<ASTIdentifier>(String(last_begin, last_end));
         }
-        else
+        else if ((!pos.isValid() && pos->type == TokenType::EndOfStream))
```
Comment (nickitat) [inline src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp:64]:
```suggestion
        else if (!pos.isValid() && pos->type == TokenType::EndOfStream)
```

**Review by nickitat — APPROVED** [review]
lgtm

File: src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp (hunk 1 above)
Comment (KevinyhZou) [reply src/Functions/JSONPath/Parsers/ParserJSONPathMemberAccess.cpp]: done

**Comment by KevinyhZou** [comment]
cc @nickitat

**Pull request merged**